be/src/runtime/fe_based_rowset_id_generator.cpp

```diff
@@ -0,0 +1,108 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+#include <string>
+#include <fstream>
+#include <thrift/Thrift.h>
+#include <thrift/transport/TSocket.h>
+#include <thrift/transport/TBufferTransports.h>
+#include <thrift/protocol/TBinaryProtocol.h>
+#include <thrift/TApplicationException.h>
+#include "gen_cpp/FrontendService.h"
+#include "gen_cpp/MasterService_types.h"
+#include "gen_cpp/HeartbeatService_types.h"
+#include "runtime/client_cache.h"
+
+#include "runtime/fe_based_rowset_id_generator.h"
+
+namespace doris {
+
+using apache::thrift::TException; 
+using apache::thrift::transport::TTransportException;
+
+FeBasedRowsetIdGenerator::FeBasedRowsetIdGenerator(ExecEnv* exec_env) : 
+    _exec_env(exec_env) {
+}
+
+OLAPStatus FeBasedRowsetIdGenerator::get_next_id(RowsetId* gen_rowset_id) {
+    std::lock_guard<std::mutex> l(_lock);
```

Comment:
std::mutex is too heavy for get next id. In most cases, it is a only add operation

Reply:
Done